File: src/main/java/com/seedsanskriti/service/AuthService.java
package com.seedsanskriti.service;

import com.seedsanskriti.dto.ForgotPasswordRequest;
import com.seedsanskriti.dto.LoginRequest;
import com.seedsanskriti.dto.RegisiterCustomerRequest;
import com.seedsanskriti.dto.RegisterSupplierRequest;
import com.seedsanskriti.dto.ResetPasswordRequest;

public interface AuthService {
	
	
	String registerCustomer(RegisiterCustomerRequest request);
	
	String registerSupplier(RegisterSupplierRequest request);
	
	String login(LoginRequest request);

	String forgotPassword(ForgotPasswordRequest request);

	String resetPassword(ResetPasswordRequest request);
	    
	}

File: frontend/src/services/authService.js
import apiClient from './apiClient';

const authService = {
  registerCustomer: (payload) =>
    apiClient.post('/auth/register/customer', payload).then((res) => res.data),

  registerSupplier: (payload) =>
    apiClient.post('/auth/register/supplier', payload).then((res) => res.data),

  // Backend returns the raw JWT string as the response body.
  login: (payload) =>
    apiClient
      .post('/auth/login', payload, { transformResponse: [(data) => data] })
      .then((res) => res.data),

  forgotPassword: (payload) =>
    apiClient.post('/auth/forgot-password', payload).then((res) => res.data),

  resetPassword: (payload) =>
    apiClient.post('/auth/reset-password', payload).then((res) => res.data),
};

export default authService;
